### PR TITLE
Add CSV Download frontend and backend

### PR DESCRIPTION
**Pull request by jojortz**
- Updated save_to_csv function for both the rag_database and qa_database
- Added endpoints for saving to CSV for RAG table and QA Table
- Created Modal UI component and custom DownloadModal for CSV filename input on download
- Added DownloadModal and Download button to Chatbot and RAGChatbot frontend compoenents

### CODE REVIEW

**Comment by goldmermaid** [comment]
Great work! I just tested the Download button, it can download all my data. 

One more feature to improve the user experience: When I downloaded the file, I didn't know where it's saved at. Could you add the note in the download UI that  "will be saved in the `~/pykoi/` folder"?

**Comment by CambioML** [comment]
Regarding the download feature, I am just curious how hard it is to have a pop up window and let user pick the download location? User might git clone pykoi at any arbitrary location, so download at `~/pykoi/` might not be a good idea.

File: pykoi/chat/db/constants.py [hunk 1]
```diff
@@ -26,5 +26,27 @@
...
+    RAG_CSV_HEADER_EDITED,
+    RAG_CSV_HEADER_VOTE_STATUS,
+    RAG_CSV_HEADER_RAG_SOURCES,
+    RAG_CSV_HEADER_SOURCE,
+    RAG_CSV_HEADER_SOURCE_CONTENT,
+    RAG_CSV_HEADER_TIMESTAMP,
+)
+
 # list separator
-QA_LIST_SEPARATOR = "||"
```
Comment (CambioML) [inline pykoi/chat/db/constants.py:30]: nit: run `black` command or add an empty line.

**Comment by CambioML** [comment]
LGTM overall.

TODO for next pr
- add saving path configuration
- ask for whether to override or not if file already exist.

**Pull request merged**